packages/notte-sdk/src/notte_sdk/types.py

```
@@ -1323,28 +1305,40 @@ def convert_selector(cls, value: str | NodeSelectors | None) -> NodeSelectors |
-                    action["id"] = ""  # TODO: find a better way to handle this
-                return ActionValidation.model_validate({"action": action}).action
+        if isinstance(action, BaseAction):
+            # already a valid action
             return action
-
+        if isinstance(action, dict):
+            if "selector" in action and "id" not in action:
+                action["id"] = ""  # TODO: find a better way to handle this
+            return ActionValidation.model_validate({"action": action}).action
+        if data is None and action is None:
+            raise ValueError("No action provided")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)